src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -497,22 +502,52 @@ public Exception logNonBanditAction(
 
             String variationValue = assignmentResult.getVariation().getTypedValue().toString();
 
+            Map<String, Double> subjectNumericAttributes = numericAttributes(subjectAttributes);
+            Map<String, String> subjectCategoricalAttributes = categoricalAttributes(subjectAttributes);
+            Map<String, Double> actionNumericAttributes = numericAttributes(actionAttributes);
+            Map<String, String> actionCategoricalAttributes = categoricalAttributes(actionAttributes);
+
             this.eppoClientConfig.getBanditLogger().logBanditAction(new BanditLogData(
               assignmentResult.getExperimentKey(),
               variationValue,
               subjectKey,
-              subjectAttributes,
               actionString,
-              actionAttributes,
               null,
-              null
+              null,
+              subjectNumericAttributes,
+              subjectCategoricalAttributes,
+              actionNumericAttributes,
+              actionCategoricalAttributes
             ));
         } catch (Exception ex) {
             loggingException = ex;
         }
         return loggingException;
     }
 
+    private Map<String, Double> numericAttributes(EppoAttributes eppoAttributes) {
+        if (eppoAttributes == null) {
+            return Map.of();
+        }
+        return eppoAttributes.entrySet().stream().filter(e -> e.getValue().isNumeric()
+        ).collect(Collectors.toMap(
+          Map.Entry::getKey,
+          e -> e.getValue().doubleValue())
+        );
+    }
+
+    private Map<String, String> categoricalAttributes(EppoAttributes eppoAttributes) {
+        if (eppoAttributes == null) {
+            return Map.of();
+        }
+        return eppoAttributes.entrySet().stream().filter(e -> !e.getValue().isNumeric() && !e.getValue().isNull()
```

Comment:
Same considerations as in the [previous PR](https://github.com/Eppo-exp/java-server-sdk-bandit-beta/pull/4#discussion_r1488657718). If my understanding is correct, we should remove the `!e.getValue().isNumeric()` condition.

Reply:
I think we need to know if the value was initialized with a truly numeric value and use that.